py3status/modules/moc.py

```diff
@@ -149,15 +154,15 @@ def on_click(self, event):
         button = event['button']
         if button == self.button_pause:
             if self.state == 'STOP':
-                self.py3.command_run('mocp --play')
+                self.py3.command_run(['mocp', '--play'])
             else:
-                self.py3.command_run('mocp --toggle-pause')
+                self.py3.command_run(['mocp', '--toggle-pause'])
         elif button == self.button_stop:
-            self.py3.command_run('mocp --stop')
+            self.py3.command_run(['mocp', '--stop'])
         elif button == self.button_next:
-            self.py3.command_run('mocp --next')
+            self.py3.command_run(['mocp', '--next'])
         elif button == self.button_previous:
-            self.py3.command_run('mocp --prev')
+            self.py3.command_run(['mocp', '--prev'])
```

Comment:
is there a reason that you have changed these?  `run_command()` will do a `shlex.split()` on the string and it's only happening on user actions so not really a performance help.  To me it just makes these harder to read.

Reply:
>so not really a performance help

Maybe not for `on_click`, but could be good for commands in `main_method`. I'm just doing them all because to avoid having to `shlex.split` repeatedly. It's a judgmental tweaks. Another one I did is to check `if self.thresholds` first before dropping numbers in. Really, Idk. :-)

Reply:
> I'm just doing them all because to avoid having to shlex.split repeatedly.

don't get too obsessed.  The optimization stuff is really hard.  Generally unless you are doing lots of profiling to check things it often makes things worse.  The most important thing by far is to have easy readable code.  There are some parts of py3status we want to improve but it is things like in the formatter and even there readability is most important and also we are talking of shaving nanoseconds (which is hardly worth it)

Most modules aren't run very often and even when they are it'll be the formatter using most of the cpu or just the `moc` call here.  so stuff like the reducing calls to `xrandr` in your other PRs is the real wins.